scene/3d/sprite_3d.cpp

```diff
@@ -421,9 +422,11 @@ SpriteBase3D::~SpriteBase3D() {
 ///////////////////////////////////////////
 
 void Sprite3D::_draw() {
-	set_base(RID());
-
+	if (get_base() != get_mesh()) {
+		set_base(get_mesh());
+	}
 	if (!texture.is_valid()) {
+		set_base(RID());
 		return;
 	}
```

Comment:
Probably a silly question, but what happens if the mesh is valid but not the texture, does it get into a loop of `set_base(mesh)` then `set_base(RID())`? If so is this problematic? I'm not really familiar with the `set_base` stuff. :blush:

Reply:
The mesh should always be valid! It is created in the beginning and used throughout. There is no way to set or remove a mesh. So no risk of getting into a loop.

Reply:
I guess we could add ``&& texture.is_valid()`` to the first conditional so that it only resets the base when a new texture is set.